learning_resources_search/views.py

```diff
@@ -74,11 +84,97 @@ def get(self, request):
             return Response(errors, status=400)
 
 
+class UserSearchSubscriptionViewSet(viewsets.ModelViewSet):
+    """
+    View for listing percolate query subscriptions for a user
+    """
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = PercolateQuerySerializer
+
+    def get_queryset(self):
+        """
+        Generate a QuerySet for fetching valid PercolateQueries for this user
+
+        Returns:
+            QuerySet of PercolateQuery objects subscribed to by request user
+        """
+        queryset = self.request.user.percolate_queries.all()
+        for backend in list(self.filter_backends):
+            queryset = backend().filter_queryset(self.request, queryset, view=self)
+        return queryset
+
+    @action(detail=False, methods=["post"], name="Subscribe user to query")
+    @extend_schema(
+        summary="Subscribe user to query",
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=PercolateQuerySerializer(),
+    )
+    def subscribe(self, request, *args, **kwargs):  # noqa: ARG002
+        request_data = LearningResourcesSearchRequestSerializer(data=request.data)
+        if request_data.is_valid():
+            percolate_query = subscribe_user_to_search_query(
+                request.user, request_data.data | {"endpoint": LEARNING_RESOURCE}
+            )
+            return Response(PercolateQuerySerializer(percolate_query).data)
+        else:
+            errors = {}
+            for key, errors_obj in request_data.errors.items():
+                if isinstance(errors_obj, list):
+                    errors[key] = errors_obj
+                else:
+                    errors[key] = list(set(chain(*errors_obj.values())))
+            return Response(errors, status=400)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        url_path="unsubscribe",
+        name="Unsubscribe user from query by id",
+    )
+    @extend_schema(
+        summary="Unsubscribe user to query",
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=PercolateQuerySerializer(),
```

Comment:
I think you can add a `description` kwarg here... Right now all the descriptions are
```
View for listing percolate query subscriptions for a user
```